packages/PhpAttribute/AnnotationToAttributeMapper/ArrayAnnotationToAttributeMapper.php

```diff
@@ -34,23 +37,41 @@ public function isCandidate(mixed $value): bool
     /**
      * @param mixed[] $value
      */
-    public function map($value): array|Expr
+    public function map($value): Expr
     {
-        $values = array_map(fn ($item): Expr|array => $this->annotationToAttributeMapper->map($item), $value);
+        $arrayItems = [];
 
-        foreach ($values as $key => $value) {
-            // remove the key and value? useful in case of unwrapping nested attributes
-            if (! $this->isRemoveArrayPlaceholder($value)) {
+        foreach ($value as $key => $singleValue) {
```

Comment:
This is the important change. Before, if the array of items was provided, only the array value was converted, not the key.
Now both key and item are converted to `ArrayItem` expr :+1: